tests/suite/test_app_protect.py

```diff
@@ -196,18 +199,50 @@ def test_responses_malformed_block(
         assert resp_valid.status_code == 200
 
         print("----------------------- Send valid request with body ----------------------")
-        headers = {
-             "Content-Type": "application/json",
-             "host": backend_setup.ingress_host}
+        headers = {"Content-Type": "application/json", "host": backend_setup.ingress_host}
         resp_valid = requests.post(backend_setup.req_url, headers=headers, data="{}", verify=False)
         print(resp_valid.text)
         assert valid_resp_addr in resp_valid.text
         assert valid_resp_name in resp_valid.text
         assert resp_valid.status_code == 200
 
         print("---------------------- Send invalid request ---------------------")
-        resp_invalid = requests.post(backend_setup.req_url, headers=headers, data="{{}}", verify=False,)
+        resp_invalid = requests.post(
+            backend_setup.req_url,
+            headers=headers,
+            data="{{}}",
+            verify=False,
+        )
         print(resp_invalid.text)
         assert invalid_resp_title in resp_invalid.text
         assert invalid_resp_body in resp_invalid.text
         assert resp_invalid.status_code == 200
+
+    @pytest.mark.parametrize("backend_setup", [{"policy": "csrf"}], indirect=True)
+    def test_responses_csrf(
+        self, kube_apis, ingress_controller_endpoint, crd_ingress_controller_with_ap, backend_setup, test_namespace
+    ):
+        """
+        Test CSRF (Cross Site Request Forgery) AppProtect policy: Block requests with invalid/null/non-https origin-header
+        """
+        print("------------- Run test for AP policy: CSRF --------------")
+        print(f"Request URL without CSRF protection: {backend_setup.req_url}")
+        print(f"Request URL with CSRF protection: {backend_setup.req_url_2}")
+
+        ensure_response_from_backend(backend_setup.req_url_2, backend_setup.ingress_host, check404=True)
+
+        print("----------------------- Send request with http origin header ----------------------")
+
+        headers = {"host": backend_setup.ingress_host, "Origin": "http://appprotect.example.com"}
+        resp_valid = requests.post(backend_setup.req_url, headers=headers, verify=False, cookies={"flavor": "darkchoco"})
```

Comment:
are cookies necessary? the policy doesn't configure any rules for cookies, right?

Reply:
@pleshakov cookies are necessary for CSRF protection to kick in